### ISSUE TITLE
train_uav1

### ISSUE THREAD

**Issue by BigdateAB** [mzzcdf/Thermal3DGS#8]
Hi I am reporting the following error while training with your dataset, what is causing this? 

![Image](https://github.com/user-attachments/assets/356ef2dd-1eff-4104-91fa-9817552076a0)
python train.py -s <path to COLMAP or NeRF Synthetic dataset>Here the s is a colmap scene ，If you specify a folder that contains more than one scene, it will also report an error。

**Comment by mzzcdf** [comment]
Thank you for your question. We re-tested the Drone1 scene based on your issue and checked the tensor shape before self.conv = nn.Conv2d(in_channels=2, out_channels=1, kernel_size=1, stride=1, padding=0) in TCM's ReshapeConv. The output shape was [2, 473, 710], and no errors occurred after applying self.conv(). Please check if your self.conv has been modified, which might have changed its input dimension requirements.

**Comment by BigdateAB** [comment]
> Thank you for your question. We re-tested the Drone1 scene based on your issue and checked the tensor shape before self.conv = nn.Conv2d(in_channels=2, out_channels=1, kernel_size=1, stride=1, padding=0) in TCM's ReshapeConv. The output shape was [2, 473, 710], and no errors occurred after applying self.conv(). Please check if your self.conv has been modified, which might have changed its input dimension requirements.

Thanks for your answer, I didn't make any changes to the code, self.conv = nn.Conv2d(in_channels=2, out_channels=1, kernel_size=1, stride=1, padding=0) in TCM's ReshapeConv, which requires 4 dimensions for its inputs Why is it that your input dimensions of [2, 473, 710] are the same as mine, yet it works fine, while mine reports an error? Is it possible that you have made changes to the code?

**Comment by mzzcdf** [comment]
Thank you for your further question! After a more thorough investigation, we believe the issue may be due to differences in the automatic broadcasting mechanism of nn.Conv2d() across different PyTorch environments. To address this, we have added a version of nn.Conv2d() without automatic broadcasting in the TCM_utils.py file, which we hope will resolve your input dimension issue.

**Comment by BigdateAB** [comment]
> Thank you for your further question! After a more thorough investigation, we believe the issue may be due to differences in the automatic broadcasting mechanism of nn.Conv2d() across different PyTorch environments. To address this, we have added a version of nn.Conv2d() without automatic broadcasting in the TCM_utils.py file, which we hope will resolve your input dimension issue.

Thank you for your answers and your colorful work!
